### ISSUE TITLE
Federation

### ISSUE THREAD

**Issue by caniko** [k8ssandra/k8ssandra-operator#505]
**What is missing?**
TLDR: Authorization system for the federation of K8ssandra data planes.

Federation requires access provision, and management. This feature is curently lacking in K8ssandra (correct me if I am wrong). The access can be delegated by a private/public key scheme where the signatures are passed to the control-plane from the federated data-plane. These private keys can be rotated automatically using something like a [transit secret engine](https://www.vaultproject.io/docs/secrets/transit) for security assurance.

**Why do we need it?**
Projects that require networks of individual groups rather than a single hierarchial entity such as a company can benefit from the horizontal scaling capabilities of Cassandra.

Cassandra alone can't and shouldn't support this feature, it should be something that is provided at the level of orchestration, K8ssandra.

**Comment by burmanm** [comment]
I might have misunderstood your request here, but to me this feels like a feature at the Kubernetes level rather than k8ssandra-operator level. For example, Openshift has security features that would allow restricting the access to those namespaces where the control-plane is deployed to with different authentication / authorisation methods.

**Comment by caniko** [comment]
> I might have misunderstood your request here, but to me this feels like a feature at the Kubernetes level rather than k8ssandra-operator level.

An eye-openening point! I am not looking for any namespace restriction, maybe this is applicable to some, but restriction to what clusters can enter the federation. I realize that I was requesting a bidirectional control between the control plane and [KubeFed](https://github.com/kubernetes-sigs/kubefed).

The authentication should be at the level of Kubernetes for sure. And scaling should be handled by the contorl plane like any other scenario. I am closing this issue.
